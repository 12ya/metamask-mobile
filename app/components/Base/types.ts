export interface TextProps {
  /**
   * Resets default style
   */
  reset?: boolean;
  /**
   * Align text to center
   */
  centered?: boolean;
  /**
   * Align text to right
   */
  right?: boolean;
  /**
   * Makes text bold
   */
  bold?: boolean;
  /**
   * Makes text green
   */
  green?: boolean;
  /**
   * Makes text black
   */
  black?: boolean;
  /**
   * Makes text blue
   */
  blue?: boolean;
  /**
   * Makes text grey
   */
  grey?: boolean;
  /**
   * Makes text red
   */
  red?: boolean;
  /**
   * Makes text orange
   */
  orange?: boolean;
  /**
   * Makes text fontPrimary color
   */
  primary?: boolean;
  /**
   * Makes text muted color
   */
  muted?: boolean;
  /**
   * Makes text italic and tight
   * used in disclaimers
   */
  disclaimer?: boolean;
  /**
   * Makes text black and bigger
   * Used in modals
   */
  modal?: boolean;
  /**
   * Makes text with bigger line height
   * Used in modals with information text
   */
  infoModal?: boolean;
  /**
   * Makes text small
   */
  small?: boolean;
  /**
   * Makes text big
   */
  big?: boolean;
  /**
   * Makes text even bigger
   */
  bigger?: boolean;
  /**
   * Makes text uppercase
   */
  upper?: boolean;
  /**
   * Applies a link style
   */
  link?: boolean;
  /**
   * Applies a strikethrough decoration
   */
  strikethrough?: boolean;
  /**
   * Applies a underline decoration
   */
  underline?: boolean;
  /**
   * Removes the vertical margin
   */
  noMargin?: boolean;
  /**
   * Any other external style defined in props will be applied
   */
  style?: any;
}
