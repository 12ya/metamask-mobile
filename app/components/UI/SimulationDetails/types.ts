import { Hex } from '@metamask/utils';
import { BigNumber } from 'bignumber.js';

export enum AssetType {
  Native = 'NATIVE',
  ERC20 = 'ERC20',
  ERC721 = 'ERC721',
  ERC1155 = 'ERC1155',
}

export const NATIVE_ASSET_IDENTIFIER: NativeAssetIdentifier = {
  type: AssetType.Native,
};

/**
 * Describes an amount of fiat.
 */
export const FIAT_UNAVAILABLE = null;
export type FiatAmountAvailable = number;
export type FiatAmount = FiatAmountAvailable | typeof FIAT_UNAVAILABLE;

/**
 * Identifies the native asset of a chain.
 */
export interface NativeAssetIdentifier {
  type: AssetType.Native;
  address?: undefined;
  tokenId?: undefined;
}

/**
 * Uniquely identifies a token asset on a chain.
 */
export interface TokenAssetIdentifier {
  type: Exclude<AssetType, AssetType.Native>;
  address: Hex;
  tokenId?: Hex;
}

export type AssetIdentifier = Readonly<
  NativeAssetIdentifier | TokenAssetIdentifier
>;

/**
 * Describes a change in an asset's balance to a user's wallet.
 */
export type BalanceChange = Readonly<{
  /**
   * The asset identifier for the balance change.
   */
  asset: AssetIdentifier;

  /**
   * The quantity of asset tokens, expressed as a decimal value.
   *
   * This property represents the amount of tokens, taking into account the
   * number of decimals supported by the asset. The value can be positive
   * (increase) or negative (decrease).
   *
   * Example: If an asset supports 18 decimals, an `amount` of 1.5 represents
   * 1.5 tokens, or more precisely, 1.5 * 10^18 of the smallest divisible unit.
   */
  amount: BigNumber;
}>;

/** Token standards supported by simulation. */
export enum SimulationTokenStandard {
  erc20 = 'erc20',
  erc721 = 'erc721',
  erc1155 = 'erc1155',
}

/** Simulation data concerning an updated token. */
export interface SimulationToken {
  /** The token's contract address. */
  address: Hex;

  /** The standard of the token. */
  standard: SimulationTokenStandard;

  /** The ID of the token if supported by the standard. */
  id?: Hex;
}

/** Simulation data concerning an update to a native or token balance. */
export interface SimulationBalanceChange {
  /** The balance before the transaction. */
  previousBalance: Hex;

  /** The balance after the transaction. */
  newBalance: Hex;

  /** The difference in balance. */
  difference: Hex;

  /** Whether the balance is increasing or decreasing. */
  isDecrease: boolean;
}

/** Simulation data concerning a change to the a token balance. */
export type SimulationTokenBalanceChange = SimulationToken &
  SimulationBalanceChange;

export enum SimulationErrorCode {
  ChainNotSupported = 'chain-not-supported',
  Disabled = 'disabled',
  InvalidResponse = 'invalid-response',
  Reverted = 'reverted',
}

/** Error data for a failed simulation. */
export interface SimulationError {
  /** Error code to identify the error type. */
  code?: string | number;

  /** Error message to describe the error. */
  message?: string;
}

/** Simulation data for a transaction. */
export interface SimulationData {
  /** Error data if the simulation failed or the transaction reverted. */
  error?: SimulationError;

  /** Data concerning a change to the user's native balance. */
  nativeBalanceChange?: SimulationBalanceChange;

  /** Data concerning a change to the user's token balances. */
  tokenBalanceChanges: SimulationTokenBalanceChange[];
}
