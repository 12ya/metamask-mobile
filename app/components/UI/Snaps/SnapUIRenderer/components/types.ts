import { JSXElement, MaybeArray } from '@metamask/snaps-sdk/jsx';

export interface UIComponentParams<T extends JSXElement> {
  map: Record<string, number>;
  element: T;
  form?: string;
}

export interface UIComponent {
  element: string;
  props?: Record<string, unknown>;
  children?: MaybeArray<UIComponent | string>;
  key?: string;
}

export type UIComponentFactory<T extends JSXElement> = (
  params: UIComponentParams<T>,
) => UIComponent;
