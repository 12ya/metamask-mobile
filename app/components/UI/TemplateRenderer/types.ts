import { SafeComponentListValues } from './SafeComponentList';

export type Sections = string | SectionShape | (string | SectionShape)[];

export interface SectionShape {
  element: keyof SafeComponentListValues;
  key: string;
  props?: Record<string, unknown>;
  children?: Sections;
}
