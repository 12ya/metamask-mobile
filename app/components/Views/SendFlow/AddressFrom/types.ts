export interface SFAddressFromProps {
    accountAddress: string;
    accountName: string;
    accountBalance: string | undefined;
    updateAccountInfo: ({accountAddress, accountName, accountBalance, balanceIsZero}: {accountAddress: string, accountName: string, accountBalance: string, balanceIsZero: boolean} ) => void;
}

export interface SelectedAsset {
    address: string;
    isETH: boolean;
    logo: string;
    name: string;
    symbol: string;
}