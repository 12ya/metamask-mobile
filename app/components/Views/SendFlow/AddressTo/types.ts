export interface SFAddressToProps {
  inputRef: any;
  highlighted: boolean;
  addressToReady: boolean;
  toSelectedAddress: any;
  toSelectedAddressName: any;
  onSubmit: (address: string) => void;
  inputWidth: any;
  confusableCollectionArray: any;
  isFromAddressBook: any;
  updateParentState: any;
}
