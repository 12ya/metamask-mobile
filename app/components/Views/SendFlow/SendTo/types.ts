export interface SFSendToProps {
    accounts: object;
    navigation: any;
    newAssetTransaction: () => void;
    selectedAddress: string;
    identities: object;
    ticker: string;
    setRecipient: () => void;
    setSelectedAsset: () => void;
    showAlert: () => void;
    providerType: string;
    route: object;
    isPaymentRequest: boolean;
    addRecent: () => void;
    frequentRpcList: object;
    addressBook: object;
    chainId: string;
    network: string;
}