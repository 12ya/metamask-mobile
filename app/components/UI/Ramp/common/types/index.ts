export type { Region } from '../../../../../redux/slices/fiatOrders/types';
export { RampType } from '../../../../../redux/slices/fiatOrders/types';

export enum PROVIDER_LINKS {
  HOMEPAGE = 'Homepage',
  PRIVACY_POLICY = 'Privacy Policy',
  SUPPORT = 'Support',
}

export * from './analytics';
