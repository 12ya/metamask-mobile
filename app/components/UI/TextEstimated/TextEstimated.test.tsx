// Third party dependencies.
import React from 'react';
import { render } from '@testing-library/react-native';

// External dependencies.
import { TextVariant } from '../../../component-library/components/Texts/Text/Text.types';

// Internal dependencies.
import TextEstimated from './TextEstimated';
import {
  TEXT_ESTIMATED_TEST_ID,
  TEST_SAMPLE_TEXT,
} from './TextEstimated.constants';

describe('TextEstimated - Snapshot', () => {
  it('should render default settings correctly', () => {
    const { toJSON } = render(
      <TextEstimated variant={TextVariant.HeadingSMRegular}>
        {TEST_SAMPLE_TEXT}
      </TextEstimated>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

describe('TextEstimated', () => {
  it('should render TextEstimated', () => {
    const { toJSON } = render(
      <TextEstimated variant={TextVariant.HeadingSMRegular}>
        {TEST_SAMPLE_TEXT}
      </TextEstimated>,
    );
    const TextEstimatedComponent = wrapper.findWhere(
      (node) => node.prop('testID') === TEXT_ESTIMATED_TEST_ID,
    );
    expect(TextEstimatedComponent.exists()).toBe(true);
  });
});
