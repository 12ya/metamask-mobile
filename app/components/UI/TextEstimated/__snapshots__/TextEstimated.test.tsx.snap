// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEstimated - Snapshot should render default settings correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
  testID="text-with-prefix-icon"
>
  <SvgMock
    color="#24272A"
    height={12}
    name="Tilde"
    style={
      {
        "height": 12,
        "width": 12,
      }
    }
    testID="text-with-prefix-icon-icon"
    width={12}
  />
  <Text
    style={
      {
        "color": "#24272A",
        "fontFamily": "Euclid Circular B",
        "fontSize": 16,
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 24,
      }
    }
    testID="text-estimated"
  >
    Sample Text
  </Text>
</View>
`;
